mixer/adapter/solarwinds/appoptics/client.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package appoptics
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"istio.io/istio/mixer/pkg/adapter"
+)
+
+// ServiceAccessor defines an interface for talking to  via domain-specific service constructs
+type ServiceAccessor interface {
+	// MeasurementsService implements an interface for dealing with  Measurements
+	MeasurementsService() MeasurementsCommunicator
+}
+
+const (
+	// MeasurementPostMaxBatchSize defines the max number of Measurements to send to the API at once
+	MeasurementPostMaxBatchSize = 1000
```

Comment:
Should this be configurable via the adapter config ?

Reply:
We could. As of today this is the max number of metrics app optics accepts at once. I will confirm with the AppOptics team to make this configurable.